couchdb-cluster-playground/steps.py

```diff
@@ -0,0 +1,152 @@
+import os
+import subprocess
+import time
+import threading
+
+from couchdbkit import Server
+import requests
+
+import lib.couchdb_custom as couch_info
+
+
+NODES = [
+    {
+        'url': 'http://localhost:5984',
+        'docker_name': 'server-0'
+    },
+    {
+        'url': 'http://localhost:15984',
+        'docker_name': 'server-1'
+    },
+    {
+        'url': 'http://localhost:25984',
+        'docker_name': 'server-2'
+    },
+]
+
+should_create_docs = False  # Boolean used for doc creation thread
+
+
+def create_docs(reporting_interval=4):
+    def create_docs_thead():
+        # thread that runs and inserts documents
+        total_count = 0
+        last_reported_time = 0
+        new_count = 0
+
+        while should_create_docs:
+            doc = {"mydoc": "test"}
+            db.save_doc(doc)
+            total_count += 1
+            new_count += 1
+            cur_time = int(time.time())
+            if ((cur_time - last_reported_time) > reporting_interval):
+                print(f'inserted {total_count} total documents, {new_count} new documents since last reported')
+                new_count = 0
+                last_reported_time = cur_time
+        print(f'insertion done: inserted {total_count} total documents')
+
+    # create database and start thread
+    server = Server("http://admin:password@localhost:5984")
+    db = server.get_or_create_db("shards")
+    x = threading.Thread(target=create_docs_thead)
+    x.start()
```

Comment:
Would creating a bunch of threads here create more load on Couch? Something like
```suggestion
    for i in range(num_threads):
        x = threading.Thread(target=create_docs_thead)
        x.start()
```

Reply:
I think that might help if we wanted to explore the concurrency of a server, but I wonder how concurrent it will be with the Python GIL. I can try this, I was going to have different threads hit all 3 servers anyway.